### DISCUSSION TITLE
ثبت مشخصات کاربر

### DISCUSSION THREAD

**Discussion by Nazari5688** [Sina-Soltani/Parbad#164] [Q&A]
سلام
میخواستم بدونم چطور میشه مشخصات کاربری که خرید رو انجام داده توی
Storage 
خود پروژه ذخیره بشه

**Comment by Sina-Soltani** [comment]
درود
منظورتون داخل استوریج خود این ابزار هست؟

**Reply by Nazari5688** [reply]
بله

**Reply by Sina-Soltani** [reply]
استوریج این ابزار در حال حاضر فقط زمانی جداگانه هست که از استوریج هایی مانند
[Cache](https://www.nuget.org/packages/Parbad.Storage.Cache/)
استفاده بکنید

اما اگر از
[EF Core](https://www.nuget.org/packages/Parbad.Storage.EntityFrameworkCore/)
استفاده می‌کنید، دیگه استوریج جداگانه نیست، بلکه درون پایگاه داده خود شما هست.
بنابراین شما تصمیم گیرنده پایگاه داده خودتون هستید

**Reply by Nazari5688** [reply]
خیلی ممنونم
من از EF Core استفاده میکنم
دیتابیس مربوط به خود ابزار اطلاعات تراکنش رو خودش ثبت میکنه
میخواستم ببینم میشه که ایدی کاربرمم به این دیتابیس اضافه کنم
یا این که کلا یک دیتابیس جداگانه باید ایجاد کنم و دستی خودم این کارو انجام بدم

**Reply by Sina-Soltani** [reply]
همونطور که گفتم این ابزار دیتابیس داخلی نداره و از دیتابیس خود شما استفاده میکنه
و خب وقتی دیتابیس مال خود شما هست و صاحبش هستید، پس بنابراین میتونید هر چیزی رو که تمایل دارین بهش اضافه کنید

اما اگر منظور شما جدول
`Table`
هست و قصد دارید فیلد جدیدی به جدول‌های
Payments, Transactions
اضافه کنید، که توسط این ابزار مقدار دهی بشه، اونوقت جواب منفی هست

هر چند که شما این امکان رو دارید که اینترفیس های استوریج رو خودتون پیاده سازی یا ویرایش کنید
[IStorage](https://github.com/Sina-Soltani/Parbad/blob/master/src/Parbad/src/Storage/Abstractions/IStorage.cs)
[IStorageManager](https://github.com/Sina-Soltani/Parbad/blob/master/src/Parbad/src/Storage/Abstractions/IStorageManager.cs)

[Using a custom storage](https://github.com/Sina-Soltani/Parbad/wiki/Configuration#using-a-custom-storage)
